tools/backups2datalad/adandi.py

```diff
@@ -236,3 +252,183 @@ async def apublish(self, max_time: float = 120) -> RemoteDandiset:
 
     async def adelete(self) -> None:
         await self.aclient.delete(self.api_path)
+
+
+class RemoteAsset(BaseModel):
+    """
+    A minimal, asynchronous version of `dandi.dandiapi.RemoteAsset`, which it
+    differs from in the following ways:
+
+    - `refetch()` method added
+    - `aclient` property added
+    - `get_digest()` and `get_raw_digest()` methods combined into
+      `get_digest_value()`
+    - No `BaseRemoteAsset` base class, as backups2datalad doesn't operate on
+      Dandiset-less asset URLs
+    - `dandiset` attribute added
+    - `dandiset_id` and `version_id` attributes are now properties
+    - `_metadata` attribute renamed to `metadata`
+    - The `metadata` attribute is now always set
+    - `from_data()` now expects ``metadata`` to be a field of ``data`` instead
+      of a separate argument
+    - The following methods & properties have been removed:
+
+      - `get_metadata()`
+      - `get_raw_metadata()` (Use the `metadata` attribute instead)
+      - `get_download_file_iter()`
+      - `download()`
+      - `set_metadata()`
+      - `set_raw_metadata()`
+      - `rename()`
+      - `delete()`
+      - `api_url`
+      - `download_url`
+
+    - `get_content_url()` does not take ``follow_redirects`` or ``strip_query``
+      parameters
+    """
+
+    dandiset: RemoteDandiset
+    #: The asset identifier
+    identifier: str = Field(alias="asset_id")
+    #: The asset's (forward-slash-separated) path
+    path: str
+    #: The size of the asset in bytes
+    size: int
+    #: The date at which the asset was created
+    created: datetime
+    #: The date at which the asset was last modified
+    modified: datetime
+    #: The asset's raw metadata
+    metadata: Dict[str, Any]
+
+    class Config:
+        allow_population_by_field_name = True
+        arbitrary_types_allowed = True
```

Comment:
```suggestion
    class Config:
        """Configuration for pydantic"""
        # To use the non-pydantic RemoteDandiset as an attribute and 
        # to be able tocreate instances with RemoteAsset(identifier=...) instead of
        # RemoteAsset(alias_id=...)
        allow_population_by_field_name = True
```